src/App.tsx

```diff
@@ -15,48 +17,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Idle = 'idle',
+  Alphabet = 'alphabet',
+  Length = 'length',
+}
+
+interface Filter {
+  active: SortType;
+  isReversed: boolean;
+}
+
+const initialFilter: Filter = {
+  active: SortType.Idle,
+  isReversed: false,
+};
+
+const getGoodsForRender = ({ active, isReversed }: Filter): string[] => {
+  const cloneGoodsFromServer = [...goodsFromServer];
+
+  switch (active) {
+    case SortType.Idle:
+      break;
+
+    case SortType.Alphabet:
+      cloneGoodsFromServer.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortType.Length:
+      cloneGoodsFromServer.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+      throw new Error('Switch case error!');
+  }
+
+  if (isReversed) {
+    cloneGoodsFromServer.reverse();
+  }
+
+  return cloneGoodsFromServer;
+};
+
 export const App: React.FC = () => {
+  const [activeFilter, setActiveFilter] = useState(initialFilter);
+
+  const isResetButtonVisible = activeFilter.active !== initialFilter.active
+    || activeFilter.isReversed !== initialFilter.isReversed;
+
+  const handleSetSortFilter = (filter: SortType) => {
+    setActiveFilter({
+      ...activeFilter,
+      active: filter,
+    });
+  };
+
+  const goodsForRender = getGoodsForRender(activeFilter);
+
+  const handleToggleReversed = () => {
+    setActiveFilter({
+      ...activeFilter,
+      isReversed: !activeFilter.isReversed,
+    });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button is-info', {
+            'is-light': activeFilter.active !== SortType.Alphabet,
+          })}
```

Comment:
If we use the classnames library, it is necessary to divide the classes into separate parts for better maintainability.
```suggestion
          className={cn('button', 'is-info', {
            'is-light': activeFilter.active !== SortType.Alphabet,
          })}
```
Fix this in all code.